File: organized/Assignements/week4/swap.py
for _ in range(int(input())):
    n=int(input())
    s=input()
    a=s
    c=0
    str(a).sort()
    for i in range(n):
        if a[i]!=s[i]:
            c+=1
    if(c>1):
        print("NO")
    else:
        print("YES")

File: organized/Assignements/week4/swap.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    int t;
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        string s;
        cin>>s;
        string a;
        a=s;
        int count=0,count1=0;
        sort(s.begin(),s.end());
        for(int i=0;i<n;i++)
        {
            if(s[i]==a[i]) count1++;
            else count++;
        }
        if(count==0 || count==2) cout<<"YES"<<endl;
        else cout<<"NO"<<endl;   
    }
    return 0;
}